RNWCPP/ReactUWP/Views/FlyoutViewManager.cpp

```diff
@@ -0,0 +1,268 @@
+  m_touchEventHanadler->AddTouchHandlers(childView);
+
+  if (m_flyout != nullptr)
+    m_flyout.Content(childView.as<winrt::UIElement>());
+}
+
+void FlyoutShadowNode::createView()
+{
+  Super::createView();
+
+  m_flyout = winrt::Flyout();
+
+  auto wkinstance = GetViewManager()->GetReactInstance();
+  m_touchEventHanadler = std::make_shared<TouchEventHandler>(wkinstance);
+
+  m_flyout.Closing([=](winrt::FlyoutBase /*flyoutbase*/, winrt::FlyoutBaseClosingEventArgs args)
+  {
+    auto instance = wkinstance.lock();
+    if (!m_updating && instance != nullptr && !m_isLightDismissEnabled && m_flyout.IsOpen())
+    {
+      args.Cancel(true);
+    }
+  });
+
+  m_flyout.Closed([=](auto&&, auto&&)
+  {
+    auto instance = wkinstance.lock();
+    if (!m_updating && instance != nullptr)
+      OnFlyoutClosed(*instance, m_tag, false);
+  });
+}
+
+/*static*/ void FlyoutShadowNode::OnFlyoutClosed(IReactInstance& instance, int64_t tag, bool newValue)
+{
+  folly::dynamic eventData = folly::dynamic::object("target", tag)("isOpen", newValue);
+  instance.DispatchEvent(tag, "topDismiss", std::move(eventData));
+}
+
+void FlyoutShadowNode::removeAllChildren()
+{
+  m_flyout.ClearValue(winrt::Flyout::ContentProperty());
+}
+
+void FlyoutShadowNode::updateProperties(const folly::dynamic&& props)
+{
+  m_updating = true;
+  bool updateTargetElement = false;
+  bool updateIsOpen = false;
+
+  if (m_flyout == nullptr)
+    return;
+
+  for (auto& pair : props.items())
+  {
+    const folly::dynamic& propertyName = pair.first;
+    const folly::dynamic& propertyValue = pair.second;
+
+    if (propertyName.asString() == "isLightDismissEnabled")
+    {
+      if (propertyValue.isBool())
+        m_isLightDismissEnabled = propertyValue.asBool();
+      else if (propertyValue.isNull())
+        m_isLightDismissEnabled = true;
+    }
+    else if (propertyName.asString() == "isOpen")
+    {
+      if (propertyValue.isBool())
+      {
+        m_isOpen = propertyValue.asBool();
+        updateIsOpen = true;
+      }
+    }
+    else if (propertyName.asString() == "placement")
+    {
+      auto placement = json_type_traits<winrt::FlyoutPlacementMode>::parseJson(propertyValue);
+      m_flyout.Placement(placement);
+    }
+    else if (propertyName.asString() == "target")
+    {
+      if (propertyValue.isInt())
+      {
+        m_targetTag = propertyValue.asInt();
+        updateTargetElement = true;
+      }
+      else
+        m_targetTag = -1;
+    }
+  }
+
+  if (updateTargetElement || m_targetElement == nullptr)
+  {
+    SetTargetFrameworkElement();
+    winrt::FlyoutBase::SetAttachedFlyout(m_targetElement, m_flyout);
+  }
+
+  if (updateIsOpen)
+  {
+    if (m_isOpen)
+    {
+      winrt::FlyoutBase::ShowAttachedFlyout(m_targetElement);
+    }
+    else
+    {
+      m_flyout.Hide();
+    }
+  }
+    //m_isOpen ? winrt::FlyoutBase::ShowAttachedFlyout(m_targetElement) : m_flyout.Hide();
+
+  Super::updateProperties(std::move(props));

```

Comment:
>Super::updateProperties(std::move(props)); [](start = 2, length = 42)

This will apply View properties to a "dummy" view which doesn't have any content.  Shouldn't this apply the view properties to m_flyout?

Reply:
You're right. This is not doing anything useful right now. All the view props will have to be hooked through FlyoutPresenterStyle in m_flyout. I'll create a separate work item for that and remove this call for now.